tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html

```diff
@@ -3138,17 +3144,65 @@ <h2>Create a distance feature</h2>
             thenDoFn(result);
           }
         });
-        this.requestManager.request(url, postData).then(wrapperFn);
+        this.requestManager.request(url, postData).then(wrapperFn)
+            .catch(reason => {
+              this.exampleStatusStr = 'Request failed'
+              self.showToast_('Request failed: ' + reason);
+              self.$.spinner.hidden = true;
+            });
+      },
+
+      /**
+       * Extract the data point feature value from an example feature. If
+       * squeeze is true then convert length 1 arrays into single objects.
+       */
+      dataPointValueFromFeature_: function(feature, squeeze) {
+        let v = null;
+        if (feature.int64List) {
+          v = feature.int64List.value;
+        } else if (feature.floatList) {
+          v = feature.floatList.value;
+        } else if (feature.bytesList && feature.bytesList.value) {
+          v = feature.bytesList.value.map(val => {
+            // Strings above a max length are most likely binary encoded data
+            // and not strings and therefore shouldn't be decoded for display
+            // in Facets Dive.
+            if (val.length > maxStringLengthToDecode) {
+              return 'ERROR: string not decoded due to length';
+            }
+            try {
+              return atob(val);
+            } catch (err) {
+              console.error('Error decoding string: ' + err);
+              return 'ERROR: error decoding string';
+            }
+          });
+        }
+        if (v) {
+          // For value lists that are of length 1, treat the feature value
+          // as just the item and not the list of length 1.
+          if (squeeze && v.length == 1) {
+            v = v[0];
+            // For numeric feature values, convert the string to a number.
+            if (!isNaN(v)) {
+              v = Number(v);
```

Comment:
```
> isNaN('a')
< true
> Number('a')
< NaN
```

I believe it is more kosher to use `parseInt(str, 10)` to cast string to number.

Reply:
I don't want to just cast ints, must handle floats too